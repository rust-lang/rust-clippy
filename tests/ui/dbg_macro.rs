// compile-flags: --test
#![warn(clippy::dbg_macro)]

fn foo(n: u32) -> u32 {
    if let Some(n) = dbg!(n.checked_sub(4)) { n } else { n }
}

fn factorial(n: u32) -> u32 {
    if dbg!(n <= 1) {
        dbg!(1)
    } else {
        dbg!(n * factorial(n - 1))
    }
}

fn main() {
    dbg!(42);
    dbg!(dbg!(dbg!(42)));
    foo(3) + dbg!(factorial(4));
    dbg!(1, 2, dbg!(3, 4));
    dbg!(1, 2, 3, 4, 5);
}

mod issue7274 {
    trait Thing<'b> {
        fn foo(&self);
    }

    macro_rules! define_thing {
        ($thing:ident, $body:expr) => {
            impl<'a> Thing<'a> for $thing {
                fn foo<'b>(&self) {
                    $body
                }
            }
        };
    }

    struct MyThing;
    define_thing!(MyThing, {
        dbg!(2);
    });
}

#[test]
pub fn issue8481() {
    dbg!(1);
}

#[cfg(test)]
fn foo2() {
    dbg!(1);
}

#[cfg(test)]
mod mod1 {
    fn func() {
        dbg!(1);
    }
}
