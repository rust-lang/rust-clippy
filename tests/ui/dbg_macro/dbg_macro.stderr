error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/auxiliary/submodule.rs:2:5
   |
LL |     dbg!();
   |     ^^^^^^^
   |
   = note: `-D clippy::dbg-macro` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::dbg_macro)]`
help: remove the invocation before committing it to a version control system
   |
LL -     dbg!();
LL +     
   |

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:9:22
   |
LL |     if let Some(n) = dbg!(n.checked_sub(4)) { n } else { n }
   |                      ^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     if let Some(n) = n.checked_sub(4) { n } else { n }
   |                      ~~~~~~~~~~~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:15:8
   |
LL |     if dbg!(n <= 1) {
   |        ^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     if n <= 1 {
   |        ~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:17:9
   |
LL |         dbg!(1)
   |         ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |         1
   |

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:20:9
   |
LL |         dbg!(n * factorial(n - 1))
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |         n * factorial(n - 1)
   |

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:26:5
   |
LL |     dbg!(42);
   |     ^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     42;
   |     ~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:28:5
   |
LL |     dbg!(dbg!(dbg!(42)));
   |     ^^^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     dbg!(dbg!(42));
   |     ~~~~~~~~~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:30:14
   |
LL |     foo(3) + dbg!(factorial(4));
   |              ^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     foo(3) + factorial(4);
   |              ~~~~~~~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:32:5
   |
LL |     dbg!(1, 2, dbg!(3, 4));
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     (1, 2, dbg!(3, 4));
   |     ~~~~~~~~~~~~~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:34:5
   |
LL |     dbg!(1, 2, 3, 4, 5);
   |     ^^^^^^^^^^^^^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     (1, 2, 3, 4, 5);
   |     ~~~~~~~~~~~~~~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:55:5
   |
LL |     dbg!();
   |     ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL -     dbg!();
LL +     
   |

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:58:13
   |
LL |     let _ = dbg!();
   |             ^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     let _ = ();
   |             ~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:60:9
   |
LL |     bar(dbg!());
   |         ^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     bar(());
   |         ~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:62:10
   |
LL |     foo!(dbg!());
   |          ^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     foo!(());
   |          ~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:64:16
   |
LL |     foo2!(foo!(dbg!()));
   |                ^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     foo2!(foo!(()));
   |                ~~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:86:9
   |
LL |         dbg!(2);
   |         ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |         2;
   |         ~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:93:5
   |
LL |     dbg!(1);
   |     ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     1;
   |     ~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:99:5
   |
LL |     dbg!(1);
   |     ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |     1;
   |     ~

error: the `dbg!` macro is intended as a debugging tool
  --> $DIR/dbg_macro.rs:106:9
   |
LL |         dbg!(1);
   |         ^^^^^^^
   |
help: remove the invocation before committing it to a version control system
   |
LL |         1;
   |         ~

error: aborting due to 19 previous errors

